src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedFieldSymbolBase.cs

```diff
@@ -33,21 +32,27 @@ internal abstract class SynthesizedFieldSymbolBase : FieldSymbol
                 (isStatic ? DeclarationModifiers.Static : DeclarationModifiers.None);
         }
 
+        internal abstract bool EmitDynamicAttributeIfDynamic
```

Comment:
The type itself doesn't have to be dynamic, it could have ```dynamic``` as one of its constituents.